File: java/shared/java/org/signal/libsignal/zkgroup/receipts/ReceiptSerial.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.libsignal.zkgroup.receipts;

import org.signal.libsignal.zkgroup.InvalidInputException;
import org.signal.libsignal.zkgroup.internal.ByteArray;

public final class ReceiptSerial extends ByteArray {

  public static final int SIZE = 16;

  public ReceiptSerial(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
  }

}

File: swift/Sources/LibSignalClient/zkgroup/ReceiptSerial.swift
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation

public class ReceiptSerial: ByteArray, @unchecked Sendable {
    public static let SIZE: Int = 16

    public required init(contents: Data) throws {
        try super.init(newContents: contents, expectedLength: ReceiptSerial.SIZE)
    }
}
